src/main/java/frc/robot/subsystems/swerve/PhoenixOdometryThread.java

```diff
@@ -116,28 +117,39 @@ public List<Samples> samplesSince(double timestamp, Set<StatusSignal<Double>> si
     }
   }
 
+  public List<Samples> samplesSince(double timestamp) {
+    return samplesSince(timestamp, signals);
+  }
+
+  public Lock getReadLock() {
+    return journalLock.readLock();
+  }
+
   @Override
   public void run() {
     while (true) {
       // Wait for updates from all signals
       var writeLock = journalLock.writeLock();
-
+      // NOTE (kevinclark): The toArray here in a tight loop is kind of ugly
+      // but keeping up a symmetric array is too and it's probably negligible on latency.
+      var status =
+          BaseStatusSignal.waitForAll(
+              2.0 / Module.ODOMETRY_FREQUENCY_HZ, signals.toArray(new BaseStatusSignal[0]));
       try {
         writeLock.lock();
-
-        // NOTE (kevinclark): The toArray here in a tight loop is kind of ugly
-        // but keeping up a symmetric array is too and it's probably negligible on latency.
-        BaseStatusSignal.waitForAll(
-            2.0 / Module.ODOMETRY_FREQUENCY_HZ, signals.toArray(new BaseStatusSignal[0]));
-        journal.add(
-            new Samples(timestampFor(signals), Maps.asMap(signals, s -> s.getValueAsDouble())));
+        if (status.isOK()) {
+          journal.add(
+              new Samples(timestampFor(signals), Maps.asMap(signals, s -> s.getValueAsDouble())));
+        } else {
+          System.out.println("Odo thread error: " + status.toString());
+        }
       } finally {
         writeLock.unlock();
       }
     }
   }
 
-  private double timestampFor(Set<BaseStatusSignal> signals) {
+  private double timestampFor(Set<StatusSignal<Double>> signals) {
```

Comment:
there was one place where we needed the specific type, and rather than make robust casting i decided to just change it to always use `StatusSignal<Double>` since thats all we use it for